projects/plugins/jetpack/extensions/blocks/premium-content/_inc/subscription-service/class-subscription-service.php

```diff
@@ -43,10 +43,11 @@ public function initialize();
 	 *
 	 * @param array  $valid_plan_ids .
 	 * @param string $access_level   .
+	 * @param int    $visitor_id   (optional) User id to use as visitor ID.
 	 *
 	 * @return boolean
 	 */
-	public function visitor_can_view_content( $valid_plan_ids, $access_level );
+	public function visitor_can_view_content( $valid_plan_ids, $access_level, $visitor_id = null );
```

Comment:
I had to change the signature again as it is used outside of request context (async-jobs), and therefore does not have a current_user

Reply:
You should `wp_set_current_user` and it will be fine. Plenty of jobs do that